internal/app/todo_app.go

```diff
@@ -0,0 +1,314 @@
+package app
+
+import (
+	"database/sql"
+	"errors"
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+)
+
+func NextDate(now time.Time, date string, repeat string) (string, error) {
+	initialDate, err := time.Parse("20060102", date)
+	if err != nil {
+		return "", errors.New("invalid date format")
+	}
+
+	if repeat == "" {
+		return "", errors.New("repeat rule is empty")
+	}
+
+	ruleParts := strings.Split(repeat, " ")
+	if len(ruleParts) == 0 {
+		return "", errors.New("invalid repeat rule")
+	}
+
+	switch ruleParts[0] {
+	case "d":
+		if len(ruleParts) != 2 {
+			return "", errors.New("invalid format for daily rule")
+		}
+		days, err := strconv.Atoi(ruleParts[1])
+		if err != nil || days <= 0 || days > 400 {
+			return "", errors.New("invalid days interval")
+		}
+
+		for {
+			initialDate = initialDate.AddDate(0, 0, days)
+			if initialDate.After(now) {
+				return initialDate.Format("20060102"), nil
+			}
+		}
+
+	case "y":
+		for {
+			initialDate = initialDate.AddDate(1, 0, 0)
+			if initialDate.After(now) {
+				return initialDate.Format("20060102"), nil
+			}
+		}
+
+	default:
+		return "", errors.New("unsupported repeat rule")
+	}
+}
+
+type Task struct {
+	ID      string `json:"id"`
+	Date    string `json:"date"`
+	Title   string `json:"title"`
+	Comment string `json:"comment,omitempty"`
+	Repeat  string `json:"repeat,omitempty"`
+}
+
+func AddTask(task *Task, db *sql.DB) (string, error) {
+	if task.Title == "" {
+		return "", errors.New("task title is required")
+	}
+
+	now := time.Now()
+	currentDay := now.Format("20060102")
+
+	if task.Date == "" || task.Date == "today" {
+		task.Date = currentDay
+	}
+
+	if _, err := time.Parse("20060102", task.Date); err != nil {
+		return "", errors.New("invalid date format, expected YYYYMMDD")
+	}
+
+	if task.Date < currentDay {
+		if task.Repeat != "" {
+			nextDate, err := NextDate(now, task.Date, task.Repeat)
+			if err != nil {
+				return "", fmt.Errorf("invalid repeat rule: %v", err)
```

Comment:
Для вывода в логи `%v`, а для оборачивания `%w` (wrap)
```suggestion
				return "", fmt.Errorf("invalid repeat rule: %w", err)
```